app/modules/favorites/components/favorite-button.tsx

```diff
@@ -0,0 +1,33 @@
+import ThumbUpIcon from '@mui/icons-material/ThumbUp';
+import ThumbUpOutlinedIcon from '@mui/icons-material/ThumbUpOutlined';
+import { IconButton } from '@mui/material';
+import { useFetcher } from '@remix-run/react';
+
+type Props = {
+  projectId: string;
+  isFavorite: boolean;
+};
+
+export function FavoriteBtn({ projectId, isFavorite }: Props) {
+  const fetcher = useFetcher();
+  const onAdd = () => fetcher.submit({ projectId }, { method: 'post' });
+  const onDelete = () => fetcher.submit({ projectId }, { method: 'delete' });
+
+  if (isFavorite) {
+    return (
+      <fetcher.Form method="delete">
+        <IconButton onClick={onDelete}>
+          <ThumbUpIcon />
+        </IconButton>
+      </fetcher.Form>
+    );
+  }
+
+  return (
+    <fetcher.Form method="post">
+      <IconButton onClick={onAdd}>
+        <ThumbUpOutlinedIcon />
+      </IconButton>
+    </fetcher.Form>
+  );
```

Comment:
You shouldn't need any `onClick` handlers
It should be enough to add hidden inputs and allow form to do the submission itself
```suggestion
  const fetcher = useFetcher();

  if (isFavorite) {
    return (
      <fetcher.Form method="delete">
        <input type="hidden" name="projectId" value={projectId} />
        <IconButton type="submit">
          <ThumbUpIcon />
        </IconButton>
      </fetcher.Form>
    );
  }

  return (
    <fetcher.Form method="post">
        <input type="hidden" name="projectId" value={projectId} />
        <IconButton type="submit">
        <ThumbUpOutlinedIcon />
      </IconButton>
    </fetcher.Form>
  );
```